python/cudf/cudf/utils/utils.py

```diff
@@ -40,6 +40,26 @@ def check_equals_float(a, b):
     )
 
 
+@njit
+def rint(x):
```

Comment:
CUDA has a `rint()` and `rintf()` function as part of its core math library (C standard math functions). 
https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__SINGLE.html#group__CUDA__MATH__SINGLE_1g7791cd93108ffc6d24524f2e8635ccfd

Before rolling your own, I think we should figure out if Numba can call this function. I note their list of python math functions supported does not include `rint`, but it does include `round()`, which hopefully calls a CUDA function.
http://numba.pydata.org/numba-doc/latest/cuda/cudapysupported.html#built-in-functions

Reply:
This is a good point @harrism . Numba's `round` was recently was fixed to succeed for integer rounding (used to segfault). It does leverage a standard CUDA function (`llrint` rather than `rint`).

We could potentially use it in place of `rint`, but not in place of the overall `round` function. 

However, when I tried this, I was not able to cleanly handle `nan_as_null=False`. In my opinion, we'd likely have to introduce more complexity than implementing `rint` to handle this scenario, which I think is less desirable.

Reply:
@beckernick what was the issue with `NaN` handling?

Reply:
@kkraus14 , it seemed that using Numab's int round with nan_as_null=False would require setting a sentinel value and then doing an additional replace on the final output to convert the sentinel to np.nan. But that would require more than just calling `s.replace(val, np.nan)`, because that replace will convert `val` into `null` regardless of whether `nan_as_null` is False or True

Reply:
Makes sense, there's no integer value for `NaN` and the binary representation of `NaN` is `min(int32/64)`. Additionally, Pandas rounds to a float whereas Numba / the CUDA APIs round to an integer which can't propagate the intermediate `NaN` values.

Reply:
There actually is a CUDA API for rounding to floats but there's no Numba binding to it unfortunately. Going to merge this and will rip it out once the libcudf function lands.

Reply:
> whereas Numba / the CUDA APIs round to an integer which can't propagate the intermediate NaN values.

@kkraus14  C/CUDA `rintf()` and `rint()` return float/double, not int. They round to nearest whole integer, but don't convert to an integer type.